lib/openhab/dsl/things.rb

```diff
@@ -96,6 +96,18 @@ def [](uid)
         alias include? []
         alias key? []
 
+        # Gets a specific thing related to a channel in the format binding_id:type_id:thing_id:channel_id
+        # or by the ChannelUID object
+        # @return Thing specified by channel or nil if the thing does not exist in the thing registry
+        def by_channel(channel_uid)
+          channel_uid = org.openhab.core.thing.ChannelUID.new(channel_uid) if channel_uid.is_a?(String)
+          thing = $things.get(channel_uid.thing_uid) # rubocop: disable Style/GlobalVars
```

Comment:
This code seems almost the same as the lookup code above in `[]`.  Is the main difference in how it gets the thing_uid?

If so, is it worth refactoring to not have the duplicate code?

Reply:
You beat me to it... I was just about to suggest the same. Also channel_uid can be nil here, in which case, we should probably call `logger.error`

Reply:
I was trying not to conflate concerns, but that shouldn't be hard.  I guess if a string was passed in I can look for more than 3 colons as a sign that we're dealing with a channel.  I'll work that in tomorrow.